packages/turbo-workspaces/__tests__/managers/npm.test.ts

```diff
@@ -0,0 +1,204 @@
+          fs.ensureDirSync(path.join(root, "node_modules"));
+        }
+
+        const project = await npm.read({
+          workspaceRoot: root,
+        });
+
+        await npm.remove({
+          project,
+          to: { name: to, version },
+          logger: new Logger({ interactive, dry }),
+          options: {
+            interactive,
+            dry,
+          },
+        });
+
+        if (dry) {
+          expect(
+            readJson<PackageJson>(project.paths.packageJson)?.workspaces
+          ).toEqual(project.workspaceData.globs);
+        } else {
+          expect(
+            readJson<PackageJson>(project.paths.packageJson)?.workspaces
+          ).toEqual(workspaces);
+        }
+      }
+    );
+  });
+
+  describe("read", () => {
+    test.each<[PackageManager, boolean]>([
+      ["yarn", true],
+      ["npm", false],
+      ["pnpm", true],
+    ])(
+      "reads npm workspaces from %s workspaces (should throw=%s)",
+      async (from, shouldThrow) => {
+        const { root, directoryName } = useFixture({
+          fixture: `../${from}/basic`,
+        });
+
+        const read = async () => npm.read({ workspaceRoot: path.join(root) });
+        if (shouldThrow) {
+          expect(read).rejects.toThrow("Not an npm workspaces project");
+          return;
+        }
+        const project = await npm.read({
+          workspaceRoot: path.join(root),
+        });
+        expect(project.name).toEqual("npm-workspaces");
+        expect(project.packageManager).toEqual("npm");
+        // paths
+        expect(project.paths.root).toMatch(
+          new RegExp(`^.*npm\/${directoryName}$`)
+        );
+        expect(project.paths.lockfile).toMatch(
+          new RegExp(`^.*npm\/${directoryName}\/package-lock.json$`)
+        );
+        expect(project.paths.packageJson).toMatch(
+          new RegExp(`^.*npm\/${directoryName}\/package.json$`)
+        );
+        // workspaceData
+        expect(project.workspaceData.globs).toEqual(["apps/*", "packages/*"]);
+        expect(project.workspaceData.workspaces).toHaveLength(4);
+        project.workspaceData.workspaces.forEach((workspace) =>
+          validateWorkspace(directoryName, workspace)
+        );
+      }
+    );
+  });
+
+  describe("convertLock", () => {
+    test.each<[PackageManager]>([["npm"], ["yarn"], ["pnpm"]])(
+      "converts %s workspaces lockfile to npm lockfile",
+      async (from) => {
+        const { root } = useFixture({ fixture: `../${from}/basic` });
+        const project = await MANAGERS[from].read({
+          workspaceRoot: root,
+        });
+
+        expect(
+          await npm.convertLock({
+            project,
+            logger: new Logger(),
+            options: {
+              interactive: false,
+              dry: false,
+            },
+          })
+        ).toBeUndefined();
+      }
+    );
+  });
+
+  describe("clean", () => {
+    test.each([
+      [true, true],
+      [false, true],
+      [false, true],
```

Comment:
Is there any reason you're running this one twice?

Reply:
Nope, swapped to deriving these.